Tests/InAppMessagingModuleSpec.swift

```diff
@@ -142,6 +158,62 @@ class InAppMessagingModuleSpec: QuickSpec {
                     }
                 }
 
+                context("and fetchAndSaveConfigData has not finished") {
+                    let queue = DispatchQueue(label: "iamTestQueue")
+
+                    beforeEach {
+                        configurationManager.fetchCalledClosure = {
+                            sleep(2)
+                        }
+                        queue.async {
+                            iamModule.initialize { }
+                        }
+                    }
+
+                    it("should wait synchronously with logEvent method") {
+                        queue.async {
+                            iamModule.logEvent(AppStartEvent())
+                        }
+                        expect(eventMatcher.loggedEvents).toAfterTimeout(beEmpty())
+                        expect(eventMatcher.loggedEvents).toEventually(haveCount(1), timeout: .seconds(2))
+                    }
+
+                    it("should wait synchronously with registerPreference method") {
+                        let preference = IAMPreferenceBuilder().setUserId("user").build()
+                        queue.async {
+                            iamModule.registerPreference(preference)
+                        }
+                        expect(preferenceRepository.preference).toAfterTimeout(beNil())
+                        expect(preferenceRepository.preference).toEventually(equal(preference), timeout: .seconds(2))
```

Comment:
Is this expect needed?
```
expect(preferenceRepository.preference).toAfterTimeout(beNil())
```

Reply:
Kind of. It checks if preference object wasn't set right after `registerPreference(preference)`. I wanted a slightly different logic to make this more understandable but I couldn't accomplish it with `waitUntil` (I'd had to use DispatchGroup)